huaweicloud/services/workspace/data_source_huaweicloud_workspace_flavors.go

```diff
@@ -0,0 +1,236 @@
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceWorkspaceFlavorsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// listFlavors: Query the list of Workspace flavors
+	var (
+		listFlavorsHttpUrl = "v2/{project_id}/products"
+		listFlavorsProduct = "workspace"
+	)
+	listFlavorsClient, err := cfg.NewServiceClient(listFlavorsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace client: %s", err)
+	}
+
+	listFlavorsPath := listFlavorsClient.Endpoint + listFlavorsHttpUrl
+	listFlavorsPath = strings.ReplaceAll(listFlavorsPath, "{project_id}", listFlavorsClient.ProjectID)
+
+	listFlavorsqueryParams := buildListFlavorsQueryParams(d)
+	listFlavorsPath += listFlavorsqueryParams
+
+	listFlavorsResp, err := pagination.ListAllItems(
+		listFlavorsClient,
+		"offset",
+		listFlavorsPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Flavor")
+	}
+
+	listFlavorsRespJson, err := json.Marshal(listFlavorsResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var listFlavorsRespBody interface{}
+	err = json.Unmarshal(listFlavorsRespJson, &listFlavorsRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("flavors", filterListFlavorsBodyFlavor(flattenListFlavorsBodyFlavor(listFlavorsRespBody), d)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListFlavorsBodyFlavor(resp interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("products", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		rst = append(rst, map[string]interface{}{
+			"id":               utils.PathSearch("product_id", v, nil),
+			"type":             utils.PathSearch("type", v, nil),
+			"architecture":     utils.PathSearch("architecture", v, nil),
+			"vcpus":            convertStrToInt(utils.PathSearch("cpu", v, nil)),
+			"memory":           utils.ConvertMemoryUnit(utils.PathSearch("memory", v, nil), 1),
+			"is_gpu":           utils.PathSearch("is_gpu", v, nil),
+			"system_disk_type": utils.PathSearch("system_disk_type", v, nil),
+			"system_disk_size": utils.PathSearch("system_disk_size", v, nil),
+			"description":      utils.PathSearch("descriptions", v, nil),
+			"charging_mode":    convertStrValue(utils.PathSearch("charge_mode", v, nil)),
+			"status":           utils.PathSearch("status", v, nil),
+		})
+	}
+	return rst
+}
+
+func convertStrValue(chargeModeCode interface{}) string {
```

Comment:
it's hard to understand from the function name.

```suggestion
func normalizeChargingMode(chargeMode interface{}) string {
```

Reply:
have changed